src/entity.rs

```diff
@@ -21,34 +22,7 @@ pub trait Entity {
 
     /// Gets a map of attributes names and values against which evaluate the
     /// entities belonging to segments.
-    fn get_attributes(&self) -> HashMap<String, AttrValue> {
+    fn get_attributes(&self) -> HashMap<String, Value> {
         HashMap::new()
     }
 }
-
-/// An attribute value can be of one of three types: numerics, strings, or
-/// booleans.
-#[derive(Debug, Clone)]
-pub enum AttrValue {
-    Numeric(f64),
```

Comment:
Using `Value` now forces the user to use `NumericValue` instead of `f64`, which is a little hard to construct. There are `FromStr` and `From<...>` implementations however.

Reply:
Don't worry too much about it, we can think about usability/ergonomics in next versions.

Reply:
Anyway, maybe the user should be explicit about the primitive type (`i64`, `u64`, `f64`), and we should keep that information to the `cmp` functions. 

This is align with your comments about

```
pub enum NumericValue{
    UnsignedInteger(u64),
    SignedInteger(i64),
    FloatingPoint(f64),
    NotRepresentable
}
```

Are we wrong trying to _merge_ `Value` and `AttrValue` into the same type? 🤔 . Or maybe our inteface should be something like the following, that forces the user to request the value with a type (aligned with `parse<T>`)?

```
let value = get_current_value::<f64>(...)?;
let value = get_current_value::<i32>(...)?;
let value = get_current_value::<String>(...)?;
...
```